server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -2331,6 +2338,80 @@ public ImageStore updateImageStoreStatus(Long id, Boolean readonly) {
         return imageStoreVO;
     }
 
+    @Override
+    public void updateStorageCapabilities(Long poolId, boolean failOnChecks) {
+        List<StoragePoolVO> pools = new ArrayList<>();
+        if (poolId == null) {
+            pools = _storagePoolDao.listByStatus(StoragePoolStatus.Up);
+        } else {
+            StoragePoolVO pool = _storagePoolDao.findById(poolId);
+
+            if (pool == null) {
+                throw new CloudRuntimeException("Primary storage not found for id: " + poolId);
+            }
+
+            pools.add(pool);
+        }
+
+        if (pools.size() == 0) {
+            throw new CloudRuntimeException("No storage pools found to update.");
+        }
+
+        for (StoragePoolVO pool: pools) {
+
+            // Only checking NFS for now - required for disk provisioning type support for vmware.
+            if (pool.getPoolType() != StoragePoolType.NetworkFilesystem) {
+                if (failOnChecks) {
```

Comment:
@Spaceman1984 how about defining `failOnChecks`locally in this method and set to true when poolId is passed, instead method parameter.

Reply:
failOnChecks is not only dependent on the pool id being null or not, it is also dependent on where the method is being called from.